src/docker/composebuilder/service_endpoints.go

```diff
@@ -46,13 +54,17 @@ func (s *ServiceEndpoints) GetPortMappings() []string {
 
 // GetEndpointMappings returns a map from env var name to env var value of a service endpoint
 func (s *ServiceEndpoints) GetEndpointMappings() map[string]string {
-	if s.ContainerPort == "" {
-		return map[string]string{}
-	}
 	switch s.ServiceConfig.Type {
 	case "public":
 		externalKey := fmt.Sprintf("%s_EXTERNAL_ORIGIN", toConstantCase(s.ServiceRole))
-		externalValue := fmt.Sprintf("http://localhost:%s", s.HostPort)
+		var externalValue string
+		if s.BuildMode.Type == BuildModeTypeLocal {
+			if s.HostPort != "" {
+				externalValue = fmt.Sprintf("http://localhost:%s", s.HostPort)
+			}
+		} else {
+			externalValue = fmt.Sprintf("https://%s.%s.com", s.ServiceRole, s.AppName)
```

Comment:
I believe this should be the app url instead of the app name. @alexdavid can you confirm. Here is the issue for reference: https://github.com/Originate/exosphere/issues/685